packages/expo-updates/src/UseUpdatesUtils.ts

```diff
@@ -0,0 +1,104 @@
+import * as Updates from './Updates';
+import type { UpdateEvent } from './Updates.types';
+import type { AvailableUpdateInfo, UpdatesInfo, UseUpdatesCallbacksType } from './UseUpdates.types';
+import { currentlyRunning } from './UseUpdatesConstants';
+
+/////// Internal functions ////////
+
+// Constructs the availableUpdate from the update manifest
+// Manifest is of type "any" until we no longer support classic updates
+export const availableUpdateFromManifest = (manifest: any) => {
+  return manifest
+    ? {
+        updateId: manifest?.id ?? null,
+        createdAt: manifest?.createdAt
+          ? new Date(manifest?.createdAt)
+          : manifest?.publishedTime
+          ? new Date(manifest?.publishedTime)
+          : null,
+        manifest,
+      }
+    : undefined;
+};
+
+// Constructs the UpdatesInfo from an event
+export const updatesInfoFromEvent = (
+  updatesInfo: UpdatesInfo | undefined,
+  event: UpdateEvent
+): UpdatesInfo => {
+  const lastCheckForUpdateTime = new Date();
```

Comment:
if we add more events to this event bus, this will need to change. it might be worth just inlining this `new Date()` in each of the created objects below to avoid accidentally applying this to future event types.

Reply:
Fixed